src/views/Project/Details/Users/SearchList/SearchListItem/index.js

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { connect } from 'react-redux';
+import { iconNames } from '#constants';
+
+import PrimaryButton from '#rsca/Button/PrimaryButton';
+
+import {
+    addProjectMembershipAction,
+    addProjectUserGroupAction,
+} from '#redux';
+
+import UserMembershipRequest from './requests/UserMembershipRequest';
+import UsergroupMembershipRequest from './requests/UsergroupMembershipRequest';
+
+import styles from './styles.scss';
+
+const propTypes = {
+    className: PropTypes.string,
+    type: PropTypes.string.isRequired,
+    memberId: PropTypes.number.isRequired,
+    projectId: PropTypes.number.isRequired,
+    username: PropTypes.string,
+    usergroupTitle: PropTypes.string,
+};
+
+const defaultProps = {
+    className: '',
+    username: undefined,
+    usergroupTitle: undefined,
+};
+
+const mapDispatchToProps = dispatch => ({
+    addProjectMember: params => dispatch(addProjectMembershipAction(params)),
+    addProjectUserGroup: params => dispatch(addProjectUserGroupAction(params)),
+});
+
+@connect(undefined, mapDispatchToProps)
+export default class SearchListItem extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+
+    constructor(props) {
+        super(props);
+
+        const setState = d => d.setState(d);
```

Comment:
what does this even mean? d.setState(d)

Reply:
LOL, this should be 
d => this.setState(d)